libs/core/src/lib/feed-input/directives/feed-input-textarea.directive.ts

```diff
@@ -0,0 +1,84 @@
+import {
+    Directive,
+    ElementRef,
+    EventEmitter,
+    HostBinding,
+    HostListener,
+    Input,
+    OnInit,
+    Output,
+    Renderer2
+} from '@angular/core';
+
+@Directive({
+    selector: '[fdFeedInputTextarea]',
+    host: {
+        '[class.fd-feed-input__textarea]': 'true',
+        '[class.fd-textarea]': 'true'
+    }
+})
+export class FeedInputTextareaDirective implements OnInit {
+    /** rows are allowed to grow */
+    @Input()
+    fdFeedInputTextareaMaxRows: number;
+
+    @HostBinding('disabled')
+    @HostBinding('attr.aria-disabled')
+    disabled: boolean;
+
+    /** Event emitted when the textarea value changed */
+    @Output()
+    valueChange = new EventEmitter<boolean>();
+
+    /** @hidden */
+    @HostListener('keyup', ['$event'])
+    onKeyup(event): void {
+        this.resize();
+        this.valueChange.emit(event.target.value);
+    }
+
+    constructor(
+        private _elementRef: ElementRef,
+        private _renderer: Renderer2
+    ) {}
+
+    /** @hidden */
+    get elementRef(): ElementRef<any> {
+        return this._elementRef;
+    }
+
+    ngOnInit(): void {
+        if (this.fdFeedInputTextareaMaxRows) {
+            const lineHeight = this._getTextareaLineHeight();
+            this._elementRef.nativeElement.style.maxHeight = lineHeight * this.fdFeedInputTextareaMaxRows + 'px';
+        }
+    }
+
+    /** @hidden make to grow textarea */
+    resize(): void {
+        this._renderer.setStyle(this.elementRef.nativeElement, 'height', 'inherit');
+
+        const totalHeight = this._getTextareaTotalHeight();
+        this._renderer.setStyle(this.elementRef.nativeElement, 'height', `${ totalHeight }px`);
+    }

```

Comment:
I see value of bringing custom solution. But could you try with `cdkTextareaAutosize` service from CDK?

Reply:
I've tried this solution already. But the solution provides a different behavior. In this case, the number of rows will depend on font size.